### PR TITLE
Handle match validation

### PR DESCRIPTION
**Pull request by ivantjh**
@moexmen/interns

### CODE REVIEW

File: app/controllers/matches_controller.rb [hunk 1]
```diff
@@ -63,9 +64,40 @@ def match_params
...
+  def check_goal (goal1, goal2)
+    @goal_check = false if goal1 == 10 && goal2 < 10
+    @goal_check = false if goal2 == 10 && goal1 < 10
+  end
+
+  def check_players (players)
+    players.each_with_index do |p|
+      other_players = players.clone
+      other_players.delete_at(players.index(p))
+      other_players.each { |op| @player_check = true if op == p}
```
Comment (sohymg) [inline app/controllers/matches_controller.rb]: `@player_check = other_players.include?(p)`

File: app/views/matches/edit.html.slim [hunk 2]
```diff
@@ -1,5 +1,7 @@
 h1.title
   | EDIT MATCH
+  - if not @error.empty?
```
Comment (sohymg) [inline app/views/matches/edit.html.slim]: `= @error if @error.present?`

Comment (sohymg) [inline app/views/matches/edit.html.slim]: generally discourage using `not`
http://stackoverflow.com/questions/27429711/can-someone-explain-not-vs-in-ruby

**Comment by weiqingtoh** [comment]
Generally speaking you can also make use of [ActiveRecord Validations](http://guides.rubyonrails.org/active_record_validations.html), which floats up errors if you use `@match.save` or `@match.update`. For example, given that you did it at the models:

``` ruby
@match = Match.find(:id)
@match.x = 'y' # update some wrong values
@match.save # returns false
@match.errors # outputs all the errors. 
```

You can give it a go at rails console to try it out.

File: app/controllers/matches_controller.rb [hunk 3]
```diff
@@ -63,9 +64,40 @@ def match_params
...
+    error[:player] = false if @player_check
+    error
+  end
+
+  def check_goal (goal1, goal2)
+    @goal_check = false if goal1 == 10 && goal2 < 10
+    @goal_check = false if goal2 == 10 && goal1 < 10
+  end
+
+  def check_players (players)
```
Comment (weiqingtoh) [inline app/controllers/matches_controller.rb]: Do you want to check if each player is unique? Or do we consider the case where 1 person can play 2 roles? (eg. @sohymg plays red_atk and red_def :stuck_out_tongue:)

**Comment by chesterhow** [comment]
Duplicate player validation not working for latest commit. @moexmen/interns

**Pull request merged**